src/read.c

```diff
@@ -799,29 +825,12 @@ static avifResult avifMetaFindOrCreateItem(avifMeta * meta, uint32_t itemID, avi
         }
     }
 
+    avifCreateItem(item);
+    AVIF_CHECKERR(*item != NULL, AVIF_RESULT_OUT_OF_MEMORY);
```

Comment:
IMPORTANT: Line 831 should be rewritten as an if statement so that we can free `*item` before returning an error:

```
    if (itemPtr == NULL) {
        avifFree(*item);
        *item = NULL;
        AVIF_RESULT_OUT_OF_MEMORY);
    }
```

Reply:
Right, I was testing, sorry about the noise.